code/modules/assembly/doorcontrol.dm

```diff
@@ -0,0 +1,153 @@
+/obj/item/device/assembly/control
+	name = "blast door controller"
+	desc = "A small electronic device able to control a blast door remotely."
+	icon_state = "control"
+	origin_tech = "magnets=1;programming=2"
+	attachable = 1
+	var/id = null
+	var/can_change_id = 0
+
+/obj/item/device/assembly/control/examine(mob/user)
+	..()
+	if(id)
+		user << "It's channel ID is '[id]'."
+
+
+/obj/item/device/assembly/control/activate()
+	spawn(0)
```

Comment:
Assembly magic. Spawn is here so sleep() wouldn't cause delays in assemblies code. I will move spawn(0) to activate() calls instead, this will fix some assembly bugs and cause some other assembly bugs.